src/main/java/com/task/service/TaskServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.task.service;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.task.dao.TaskDao;
+import com.task.dto.TaskDto;
+import com.task.model.Task;
+
+@Service
+public class TaskServiceImpl implements TaskService {
+
+	@Autowired
+	private TaskDao taskDao;
+
+	@Override
+	public List<TaskDto> selectAllTask() {
+		Iterable<Task> tasks = taskDao.findAll();
+		List<TaskDto> taskDtos = new ArrayList<TaskDto>();
+		if(tasks != null) {
+			Iterator<Task> iterator = tasks.iterator();
+			while(iterator.hasNext()) {
+				Task task = iterator.next();
+				TaskDto taskDto = new TaskDto(task.getTaskNumber(), task.getTaskName(), task.getTaskDescription(), task.isTaskStatus());
+				taskDto.setId(task.getId());
+				taskDtos.add(taskDto);
+			}
+		}
+		return taskDtos;
+	}
+
+	@Override
+	public String addTask(TaskDto taskDto) {
+		Task task = taskDao.findByTaskNumber(taskDto.getTaskNumber());
+		if(task != null)
+			return "Task Number should be unique. Please provide some unique value.";
```

Comment:
Serwis nie powinien zwracać treści odp (a co jak np będziesz chciał zmienić wersję jezykowa?). Serwis w odpowiedzi na stworzenie taska powinien zwrócić obiekt, który stworzył. To od wywołującego zależy co z tym zrobi. Powinieneś dążć do tego, żeby Twój serwis można było wykorzystać zmieniając całkowicie warstwę go wywołującą (np. na request wysłany przez http)

Reply:
Poprawilem. Teraz zwraca obiekt i rzuca exception gdy Task Number istnieje